ValidationLibrary.Rules.Tests/BaseRuleTests.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+using System.Threading.Tasks;
+using NSubstitute;
+using NUnit.Framework;
+using Octokit;
+
+namespace ValidationLibrary.Tests.Rules
+{
+    [TestFixture]
+    public abstract class BaseRuleTests<T> where T : IValidationRule
+    {
+        public User Owner { get; set; }
+        protected T _rule;
+
+        public IGitHubClient MockClient { get; set; }
+        public IRepositoriesClient MockRepositoryClient { get; set; }
+        public IRepositoryContentsClient MockRepositoryContentClient { get; set; }
+
+        public const string MainBranch = "master";
+
+        [SetUp]
+        public void Setup()
+        {
+            Owner = new User(null, null, null, 0, null, DateTime.UtcNow, DateTime.UtcNow, 0, null, 0, 0, false, null, 0, 0, null, "protacon", "protacon", null, 0, null, 0, 0, 0, null, new RepositoryPermissions(), false, null, null);
+
+            MockClient = Substitute.For<IGitHubClient>();
+            MockRepositoryClient = Substitute.For<IRepositoriesClient>();
+            MockClient.Repository.Returns(MockRepositoryClient);
+            MockRepositoryContentClient = Substitute.For<IRepositoryContentsClient>();
+            MockRepositoryClient.Content.Returns(MockRepositoryContentClient);
+            OnSetup();
+        }
+
+        protected abstract void OnSetup();
+
+        [Test]
+        public async Task Init_DoesntThrow()
+        {
```

Comment:
Just a suggestion, maybe there could be a "ThrowsWhenNull" like below and the rule classes check the client is not null